index/src/index.rs

```diff
@@ -70,27 +86,47 @@ impl Index {
         &self.index
     }
 
+    pub fn searcher(&self) -> Result<Searcher> {
+        self.reader
+            .reload()
+            .map_err(|e| ReloadError(e.to_string()))?;
+        Ok(self.reader.searcher())
+    }
```

Comment:
So the reader is updated every time the searcher is requested? That sounds good :+1: